portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,52 +26,62 @@
 import com.google.gson.Gson;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Date;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that contains saves and gets comments from database. */
+/** Servlet that contains, saves and gets comments from database. 
+ *  Uses LoginServlet to determine if a user is logged in to
+ *  either show or hide comments.
+ */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-    private List<String> comments = new ArrayList<>();
     private final Gson gson = new Gson();
+    private String currentUserEmail;
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        Query find = new Query("comment").addSort("timestamp", SortDirection.DESCENDING);
-        DatastoreService data = DatastoreServiceFactory.getDatastoreService();
-        PreparedQuery results = data.prepare(find);
+        Query query = new Query("comment").addSort("date", SortDirection.DESCENDING);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+        response.setContentType("application/json;");
 
-        //Construct an array of comment history
-        List<String> history = new ArrayList<>();
-        for (Entity entry: results.asIterable()) {
-            String curr = (String) entry.getProperty("comment");
-            history.add(curr);
-        }
+        List<Comment> commentHistory = new ArrayList<>();
+        LoginInfo userLogin = (LoginInfo) request.getAttribute("userLogin");
 
-        response.setContentType("application/json;");
-        response.getWriter().println(gson.toJson(history));
+        // Display comments if logged in
+        if(userLogin.isLoggedIn()) {
+            for (Entity entry: results.asIterable()) {
+                String content = (String) entry.getProperty("content");
+                Date date = (Date) entry.getProperty("date");
+                String email = (String) entry.getProperty("email");
+
+                commentHistory.add(new Comment(content, date, email));
+            }
+            currentUserEmail = userLogin.getUserEmail();
+        }
+        Access access = new Access(commentHistory, userLogin);
+        response.getWriter().println(gson.toJson(access));
     }
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        String text = request.getParameter("input");
-        comments.add(text);
-        long timestamp = System.currentTimeMillis();
+        String content = request.getParameter("input");
+        Date date = new Date();
+        String email = currentUserEmail;
```

Comment:
Notice that there's no user check here, so anyone, including someone who is not logged in, can post comments.